index.js

```diff
@@ -2984,37 +2984,55 @@
 
   //. ### Array
 
-  //# append :: a -> Array a -> Array a
+  //# append :: (Applicative f, Semigroup (f a)) => a -> f a -> f a
```

Comment:
> let's update the constraints in the signatures to `(Applicative f, Semigroup (f a))`

I don't understand how this differs from `(Applicative f, Semigroup f)`. Doesn't it mean exactly the same thing? If `f` is not a Semigroup, wouldn't that mean by extension that `f a` isn't a Semigroup either?